src/xspi/mod.rs

```diff
@@ -176,6 +175,28 @@ mod common {
             }
         }
     }
+    #[derive(Debug, Copy, Clone)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    pub struct XspiModes {
+        /// IO lines used for the instruction phase.
+        pub instruction: XspiMode,
+        /// IO lines used for the address phase.
+        pub address: XspiMode,
+        /// IO lines used for the alternate byte phase.
+        pub alt_byte: XspiMode,
+        /// IO lines used for the data phase.
+        pub data: XspiMode,
+    }
+    impl XspiModes {
+        pub fn new(mode: XspiMode) -> Self {
```

Comment:
```suggestion
        pub fn all(mode: XspiMode) -> Self {
```
Suggestion (I just saw this while browsing around, feel free to ignore): If this is called `all` then it is very clear that all modes are set to the given mode.

Reply:
> Suggestion (I just saw this while browsing around, feel free to ignore): If this is called `all` then it is very clear that all modes are set to the given mode.

@Abestanis Thanks for the suggestion; I'll let the maintainers decide what they prefer.

Reply:
I see the reasoning behind calling it `all`, but `new` is definitely closer to the [naming convention](https://rust-lang.github.io/api-guidelines/naming.html#c-case) and is less surprising when quickly discovering the API. Thanks for the suggestion though @Abestanis !